pkg/controller/migcluster/validation.go

```diff
@@ -346,3 +356,46 @@ func (r *ReconcileMigCluster) validateSaTokenPrivileges(cluster *migapi.MigClust
 	}
 	return nil
 }
+
+// validate operator version.
+func (r ReconcileMigCluster) validateOperatorVersionMatchesHost(cluster *migapi.MigCluster) error {
+	if cluster.Spec.IsHostCluster {
+		return nil
+	}
+	if cluster.Status.HasCriticalCondition() {
+		return nil
+	}
+
+	clusterClient, err := cluster.GetClient(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	clusterOperatorVersion, err := cluster.GetOperatorVersion(clusterClient)
+
+	if err != nil {
+		cluster.Status.SetCondition(migapi.Condition{
+			Type:     ClusterOperatorVersionNotFound,
+			Status:   True,
+			Reason:   VersionNotFound,
+			Category: Critical,
+			Message:  fmt.Sprintf("The configmap key could not be found. See logs for details."),
```

Comment:
Conditions are the first-class way we try to get information to users, so where possible it's preferable to put the information directly in the message instead of asking user to go to logs. In this case, I think we could make this something like:
```
MTC Operator version lookup failed. Key %key_name  not found in 
configmap %configmap_namespace/%configmap_name.
Ensure that all clusters are running the same MTC Operator version.
```

Reply:
As discussed, for 1.4.2 (upcoming z-stream) we should set all of these conditions to be `Warning` instead of `Critical`. Sorry for the flip-flopping.